### DISCUSSION TITLE
Where do I put IPAdapter files??

### DISCUSSION THREAD

**Discussion by EddieSoloProjects** [Stability-AI/StableSwarmUI#334] [Q&A]
Sorry for the Noob Question. I'm doing a tutorial and it wants me to put IPAdapter file into:
ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_IPAdapter_plus\models
But it looks like the default folders aren't used by StableSwarm. So where should I put them?

**Comment by mcmonkey4eva** [comment]
So, first: that's an outdated tutorial, don't follow it. To use IPAdapter in swarm... literally no installation steps needed. Just run swarm, drag an image to your prompt box, and you'll get the "ReVision" parameters on the left -- in the param box will be a button to install IPAdapter, you just click on that, accept the confirmation, and wait a minute for it to install. Once it does it'll reload the parameter view and let you select an IPAdapter model to apply. Any time you select a model that's not already downloaded, it will autodownload the model for you.

Second, for the record / for potential future relevance: If you do need to access the comfy dir, it's found under `(Swarm)/dlbackend/comfy/ComfyUI`

**Reply by EddieSoloProjects** [reply]
Well what if it's part of a Comfy UI workflow? such as this video: https://youtu.be/849xBkgpF3E

**Reply by mcmonkey4eva** [reply]
Theen
> Second, for the record / for potential future relevance: If you do need to access the comfy dir, it's found under `(Swarm)/dlbackend/comfy/ComfyUI`

Does work. But you should probably still just let Swarm install IPAdapter for you. Run any gen with it one to make sure the IPAdapter model is downloaded.
